lib/chef/knife/winrm_knife_base.rb

```diff
@@ -99,71 +109,68 @@ def create_winrm_session(options={})
           end
 
           def resolve_session_options
-            resolve_winrm_basic_options
-            resolve_winrm_auth_settings
-            resolve_winrm_kerberos_options
-            resolve_winrm_transport_options
-            resolve_winrm_ssl_options
+            @session_opts = {
+              user: resolve_winrm_user,
+              password: locate_config_value(:winrm_password),
+              port: locate_config_value(:winrm_port),
+              operation_timeout: resolve_winrm_session_timeout,
+              basic_auth_only: resolve_winrm_basic_auth,
+              disable_sspi: resolve_winrm_disable_sspi,
+              transport: resolve_winrm_transport,
+              no_ssl_peer_verification: resolve_no_ssl_peer_verification
+            }
+            if @session_opts[:transport] == :kerberos
+              @session_opts.merge!(resolve_winrm_kerberos_options)
+            end
+            @session_opts[:ca_trust_path] = locate_config_value(:ca_trust_file) if locate_config_value(:ca_trust_file)
           end
 
-          def resolve_winrm_basic_options
-            @session_opts = {}
-            @session_opts[:user] = locate_config_value(:winrm_user)
-            @session_opts[:password] = locate_config_value(:winrm_password)
-            @session_opts[:port] = locate_config_value(:winrm_port)
-
-            #30 min (Default) OperationTimeout for long bootstraps fix for KNIFE_WINDOWS-8
-            @session_opts[:operation_timeout] = locate_config_value(:session_timeout).to_i * 60 if locate_config_value(:session_timeout)
+          def resolve_winrm_user
+            user = locate_config_value(:winrm_user)
+            
+            if resolve_winrm_basic_auth ||
+              resolve_winrm_transport == :kerberos ||
+              user.include?("\\") ||
+              user.include?("@")
+              user
+            else
+              ".\\#{user}"
+            end
           end
 
-          def resolve_winrm_kerberos_options
-            if config.keys.any? {|k| k.to_s =~ /kerberos/ }
-              @session_opts[:transport] = :kerberos
-              @session_opts[:keytab] = locate_config_value(:kerberos_keytab_file) if locate_config_value(:kerberos_keytab_file)
-              @session_opts[:realm] = locate_config_value(:kerberos_realm) if locate_config_value(:kerberos_realm)
-              @session_opts[:service] = locate_config_value(:kerberos_service) if locate_config_value(:kerberos_service)
-            end
+          def resolve_winrm_session_timeout
+            locate_config_value(:session_timeout).to_i * 60 if locate_config_value(:session_timeout)
```

Comment:
Can you re-add the comment that was here? (https://github.com/chef/knife-windows/pull/282/files#diff-4a7d2c3d64141072c2155c9e8428d277L115)